### PR TITLE
SAA-1661: Add attendance required flag to attendance summary

### PR DESCRIPTION
**Pull request by munslowl**

### CODE REVIEW

File: src/main/resources/migrations/common/R__v_scheduled_instance_attendance_summary.sql [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    acts.internal_location_code,
+    acts.internal_location_description,
+    si.cancelled,
+    (SELECT COUNT(scheduled_instance_id) FROM v_prisoner_scheduled_activities psa where psa.scheduled_instance_id = si.scheduled_instance_id) AS allocations,
+    att.attendees,
+    att.not_recorded,
+    att.attended,
+    att.absences,
+    att.paid,
+    act.attendance_required
```
Comment (munslowl) [inline src/main/resources/migrations/common/R__v_scheduled_instance_attendance_summary.sql:25]: New flag

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/ActivityScheduleInstanceIntegrationTest.kt [hunk 2]
```diff
@@ -444,6 +444,7 @@ class ActivityScheduleInstanceIntegrationTest : IntegrationTestBase() {
           inCell = false,
           onWing = false,
           offWing = false,
+          attendanceRequired = true,
```
Comment (dafhoustonmoj) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/ActivityScheduleInstanceIntegrationTest.kt:447]: could be worth having an extra attendance summary record where attendance required is false?

Comment (munslowl) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsactivitiesmanagementapi/integration/ActivityScheduleInstanceIntegrationTest.kt:447]: Honestly - its just data and could spawn lots of needless tests if we go down there.

**Review by dafhoustonmoj — APPROVED** [review]
LGTM

**Pull request merged**